sdk/include/opentelemetry/sdk/http/http_client.h

```diff
@@ -0,0 +1,139 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+#include "opentelemetry/http/http_client.h"
+#include "opentelemetry/version.h"
+
+#include <map>
+#include <vector>
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace http
+{
+namespace http_api = opentelemetry::http;
+
+// HttpHeaders implementation
+class HttpHeaders : http_api::HttpHeaders, std::multimap<std::string, std::string>
+{
+
+public:
+  virtual void set(nostd::string_view const &name, nostd::string_view const &value) override
+  {
+    auto range = equal_range(std::string(name.data()));
+    auto hint  = erase(range.first, range.second);
+    insert(hint, std::make_pair(name, value));
+  }
+
+  virtual void add(nostd::string_view const &name, nostd::string_view const &value) override
+  {
+    insert(std::make_pair(name, value));
+  }
+
+  virtual nostd::string_view const &get(nostd::string_view const &name) const override
+  {
+    auto it = find(std::string(name.data()));
+    return (it != end()) ? it->second : m_empty;
+  }
+
+  virtual bool has(nostd::string_view const &name) const override
+  {
+    auto it = find(std::string(name.data()));
+    return (it != end());
+  }
+
+private:
+  nostd::string_view m_empty{};
+};
+
+class SimpleHttpRequest : public http_api::HttpRequest
+{
+
+public:
+  SimpleHttpRequest(nostd::string_view const &id) : id_(id), method_("GET") {}
+
+  // Gets the HTTP request ID.
+  virtual const nostd::string_view &GetId() const override { return id_; }
+
+  // Sets the request method
+  virtual void SetMethod(nostd::string_view const &method) override { method_ = method; }
+
+  // Sets the HTTP request URI.
+  virtual void SetUrl(nostd::string_view const &url) override { url_ = url; }
+
+  // Gets the HTTP request headers.
+  virtual http_api::HttpHeaders *GetHeaders() const override { return headers_; }
+
+  // Sets the request body.
+  virtual void SetBody(const uint8_t *const body, const size_t len) override
+  {
+    for (int i = 0; i < len; i++)
+    {
+      body_.push_back(body[i]);
+    }
+  }
+
+  virtual void GetBody(uint8_t *body, size_t &len) override
+  {
+    len = 0;
+    for (auto &e : body_)
+    {
+      body[len++] = e;
+    }
+    len = (len > 0) ? len - 1 : len;
+  }
+
+private:
+  http_api::HttpHeaders *headers_;
```

Comment:
I think we could avoid raw pointer here. Also for the getter below - maybe we should return a reference instead. Also may need to spell out that the headers multimap may only be obtained on response object upon request completion / when the callback is invoked, to prevent a race condition between request - callback - response / prevent concurrent modification of the multimap.

Reply:
done.